Add documentation for pipeline templates (EDGE-963) (#2255)
buildkite/docs: pages/pipelines/templates.md

# Pipeline templates

> 📘 Enterprise feature

> > Pipeline templates is only available on an [Enterprise](https://buildkite.com/pricing) plan.

Reviewer: Nitpick: This works when you think about "Pipeline templates" as a feature name, but I think it reads more smoothly with "are." Happy for you to decide which way to go.

```suggestion
> Pipeline templates are only available on an [Enterprise](https://buildkite.com/pricing) plan.
```

## Overview

Pipeline templates allow you to define standard pipeline step configurations to use across all the pipelines in your organization.

When a pipeline has a template assigned, the pipeline inherits its step configuration from the template. Configuring a custom step configuration for that pipeline in the Buildkite dashboard is no longer possible.

## Creating a pipeline template

Only administrators can create or update pipeline templates.

To create a template:

1. Navigate to your [organization’s pipeline templates](https://buildkite.com/organizations/-/pipeline-templates).
1. If this is your first template, select _Create a Template_. Otherwise, select _New Template_. A pipeline template is created automatically for you.
1. Update the name and description of your new template and select _Apply_ to save your changes.

> 1. You can also select _Edit_ to change the step configuration for the template.

Reviewer: I don't think the "also" fits well in the numbered steps.

```suggestion
1. Select _Edit_ if you want to change the step configuration for the template.
```
